### ISSUE TITLE
Breaks when updating svelte > 3.25.0

### ISSUE THREAD

**Issue by JackDra** [dopry/svelte-oidc#7]
I am assuming this is a known bug, considering the code is build with 3.24.0, but when upgrading svelte to 3.25.0 causes issues with the buttons.

Pressing Login or Logout causes:
```
Uncaught (in promise) Error: Function called outside component initialization
    at get_current_component (index.mjs:629)
    at getContext (index.mjs:662)
    at login (OidcContext.svelte:54)
    at HTMLButtonElement.click_handler (Login.svelte:74)
    at HTMLButtonElement.<anonymous> (index.mjs:238)
get_current_component @ index.mjs:629
getContext @ index.mjs:662
login @ OidcContext.svelte:54
click_handler @ Login.svelte:74
(anonymous) @ index.mjs:238
```

where the part which is breaking is for Login:
https://github.com/dopry/svelte-oidc/blob/a4aa8439f116a86ad281c4d0b867b3bbb02fd6ae/src/components/OidcContext.svelte#L54

and for Logout is:
https://github.com/dopry/svelte-oidc/blob/a4aa8439f116a86ad281c4d0b867b3bbb02fd6ae/src/components/OidcContext.svelte#L74

It may be related to https://stackoverflow.com/questions/60591927/svelte-user-registration-issue-with-setting-store-value, but I didn't understand it very well.

Just thought I would leave this here so it is known what is blocking any svelte updates.

**Comment by pmcneil** [comment]
Trying with new svelte app and seeing the same issue which clicking login

**Comment by Milos5611** [comment]
Any fix for this? I see the same with svelte 3.0.0

**Comment by Milos5611** [comment]
@JackDra @pmcneil have you guys found a workaround?

**Comment by JackDra** [comment]
Sorry, I never got around to implementing a fix. I think the issue is purely a svelte issue, so I think taking the code that was used here and reimplementing it from scratch would solve it

**Comment by BlackVoid** [comment]
I've made a PR that fixes the issues. It reworks the internals of the component, but should make it works with Svelte 3.32.3 and is mostly backwards compatible, unless you are using the context (which the example does not)

**Comment by dopry** [comment]
fixed in e4cf984082a11e84e4b71e519bc732bdb42ca57e
